jerry-core/parser/js/js-scanner.c

```diff
@@ -3193,7 +3193,7 @@ scanner_scan_all (parser_context_t *context_p, /**< context */
     }
     PARSER_CATCH
     {
-      JERRY_ASSERT (context_p->error == PARSER_ERR_NO_ERROR);
+      JERRY_ASSERT (context_p->error == PARSER_ERR_NO_ERROR || context_p->error == PARSER_ERR_OUT_OF_MEMORY);
```

Comment:
the `PARSER_TRY (context_p->try_buffer)` could be enclosed in `if (context_p->error != PARSER_ERR_OUT_OF_MEMORY)`

Reply:
This is not a proper fix since in the internal try  block `scanner_pop_literal_pool` may allocates memory which could lead to OOM error.